scripts/import/import_variant_synonyms

```diff
@@ -147,20 +197,337 @@ sub download_PharmGKB{
+    WHERE
+      vs.name = ? AND
+      v.variation_id=vf.variation_id
+    LIMIT 1
+  };
+  my $id_sth = $db_adaptor->dbc->prepare($id_stmt);
+  my $syn_sth = $db_adaptor->dbc->prepare($syn_stmt);
+  
+  my %mapping;
+  
+  foreach my $rs_id (@{$rs_ids}) {
+    $id_sth->bind_param(1,$rs_id,SQL_VARCHAR);
+    $id_sth->execute();
+    my ($var_id,$var_name);
+    $id_sth->bind_columns(\$var_id,\$var_name);
+    $id_sth->fetch();
+    
+    # If we couldn't find the rs_id, look in the synonym table
+    if (!defined($var_id)) {
+      $syn_sth->bind_param(1,$rs_id,SQL_VARCHAR);
+      $syn_sth->execute();
+      $syn_sth->bind_columns(\$var_id,\$var_name);
+      $syn_sth->fetch();
+    }
+    warn "$rs_id - no mapping found in variation db\n" unless $var_id ;  
+
+    $mapping{$rs_id} = [$var_id,$var_name] if $var_id && $var_name;
+  }
+  
+  return \%mapping;
+} 
+
+sub get_or_add_source {
+  my $source_name        = shift;
+  my $source_description = shift;
+  my $source_url         = shift;
+  my $source_status      = shift;
+  my $db_adaptor         = shift;
+
+  my $stmt = qq{
+    SELECT
+      source_id
+    FROM
+      source
+    WHERE
+      name = '$source_name'
+    LIMIT 1
+  };
+
+  my $sth = $db_adaptor->dbc->prepare($stmt);
+  $sth->execute();
+  my $source_id;
+  $sth->bind_columns(\$source_id);
+  $sth->fetch();
+  
+  if (!defined($source_id)) {
+    $stmt = qq{
+      INSERT INTO
+        source (
+          name,
+          description,
+          url,
+          somatic_status,
+          version
+        )
+      VALUES (
+        '$source_name',
+        '$source_description',
+        '$source_url',
+        '$source_status',
+        '$source_version'
+      )
+    };
+    $db_adaptor->dbc->do($stmt);
+    $source_id = $db_adaptor->dbc->db_handle->{'mysql_insertid'}; 
+  }
+  else {
+    $stmt = qq{
+      UPDATE
+        source 
+      SET name=?,
+        description=?,
+        url=?,
+        version=?
+      WHERE
+        source_id=?
+    };
+    my $update_source_sth =$db_adaptor->dbc->prepare($stmt);
+    $update_source_sth->execute($source_name,$source_description,$source_url,$source_version,$source_id);
+  }
+
+  return $source_id;
+}
+ 
+sub add_set_uniprot {  
+  my $db_adaptor      = shift;
+  my $set_name        = shift; 
+  my $set_description = shift;
+
+  # check if UniProt already exists 
+  my $stmt = qq{
+    SELECT
+      variation_set_id
+    FROM
+      variation_set
+    WHERE
+      name = '$set_name'
+    LIMIT 1
+  };
+
+  my $sth = $db_adaptor->dbc->prepare($stmt);
+  $sth->execute();
+  my $variation_set_id;
+  $sth->bind_columns(\$variation_set_id);
+  $sth->fetch();
+
+  if (!defined($variation_set_id)) {  
+    my $stmt_attrib = qq{
+      SELECT
+        attrib_id
+      FROM
+        attrib
+      WHERE
+        value = '$set_name'
+      LIMIT 1
+    };
+    my $sth_attrib = $db_adaptor->dbc->prepare($stmt_attrib);
+    $sth_attrib->execute();
```

Comment:
I missed this! Also, I changed the source_name from 'ph_uniprot' to 'UniProt variants'. When searching for source_name in attrib table it will probably fail because attrib.value = 'ph_uniprot'.

Reply:
I suggest a check is still needed to check that an attrib_id has been found, else it will still insert 0 if not found